post-til-virksomhet/src/main/java/no/difi/meldingsutveksling/ptv/CorrespondenceAgencyMessageFactory.java

```diff
@@ -133,96 +111,112 @@ public static InsertCorrespondenceV2 create(CorrespondenceAgencyConfiguration po
-            binaryAttachmentV2.setName(reporteeFactory.createBinaryAttachmentV2Name(a.getName()));
-            binaryAttachmentV2.setEncrypted(false);
-            binaryAttachmentV2.setSendersReference(reporteeFactory.createBinaryAttachmentV2SendersReference("AttachmentReference_as123452"));
-            binaryAttachmentV2.setData(reporteeFactory.createBinaryAttachmentV2Data(a.getData()));
-            attachmentExternalBEV2List.getBinaryAttachmentV2().add(binaryAttachmentV2);
-        });
+        if (type == null) {
+            tokens.getTextToken().add(createTextToken(0, edu.getSender().getName()));
+            tokens.getTextToken().add(createTextToken(1, serviceEditionMapping.get(Integer.valueOf(getServiceEditionCode
+                    (config).getValue()))));
+            tokens.getTextToken().add(createTextToken(2, edu.getReceiver().getName()));
+            return tokens;
+        }
 
-        AttachmentsV2 attachmentsV2 = new AttachmentsV2();
-        attachmentsV2.setBinaryAttachments(objectFactory.createAttachmentsV2BinaryAttachments(attachmentExternalBEV2List));
-        externalContentV2.setAttachments(objectFactory.createExternalContentV2Attachments(attachmentsV2));
-        correspondence.setContent(objectFactory.createMyInsertCorrespondenceV2Content(externalContentV2));
+        switch (type) {
+            case EMAIL:
+                tokens.getTextToken().add(createTextToken(0, config.getEmailSubject()));
+                tokens.getTextToken().add(createTextToken(1, config.getEmailBody()));
+                break;
+            case SMS:
+                tokens.getTextToken().add(createTextToken(0, config.getSmsText()));
+                tokens.getTextToken().add(createTextToken(1, ""));
+                break;
+            default:
+                tokens.getTextToken().add(createTextToken(0, edu.getSender().getName()));
+                tokens.getTextToken().add(createTextToken(1, serviceEditionMapping.get(Integer.valueOf(getServiceEditionCode
+                        (config).getValue()))));
+                tokens.getTextToken().add(createTextToken(2, edu.getReceiver().getName()));
+        }
 
-        no.altinn.services.serviceengine.correspondence._2009._10.ObjectFactory correspondenceObjectFactory = new no.altinn.services.serviceengine.correspondence._2009._10.ObjectFactory();
-        final InsertCorrespondenceV2 myInsertCorrespondenceV2 = correspondenceObjectFactory.createInsertCorrespondenceV2();
-        myInsertCorrespondenceV2.setCorrespondence(correspondence);
-        myInsertCorrespondenceV2.setSystemUserCode(postConfig.getSystemUserCode());
-        // Reference set by the message sender
-        myInsertCorrespondenceV2.setExternalShipmentReference(values.getExternalShipmentReference());
+        return tokens;
+    }
 
-        return myInsertCorrespondenceV2;
+    private static TextTokenSubstitutionBEList createCustomSmsTokens(CorrespondenceAgencyConfiguration config) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "createCustomSmsTokens" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AUnusedPrivateMethod)